ref/README.md

```diff
@@ -1,56 +1,160 @@
+$ make VERBOSE=1
+```
+
+On many systems, the above will suffice. However, some systems will require you to help cmake
+find dependencies, particularly if software depencencies are not installed in standard locations.
+See below for more information.
+
+### Machines that use modules to manage software
 
+Most HPC systems use modules to manage software.  Make sure you have loaded the versions of
+the compiler and software you want to use before running the build steps above.  This will allow build
+dependencies to be found properly.  For example:
+
+```bash
+$ module load intel netcdf cmake
 ```
-rm -rf build ; mkdir build ; cd build
-module load <compiler> netcdf cmake
-export CC=<gcc | icc>
-export FC=<gfortran | ifort>
-cmake -DCMAKE_BUILD_TYPE=<debug | release> ..
-make VERBOSE=1
+
+### Machines that do not use modules to manage software
+
+If compilers and/or NetCDF is not installed in a standard location where cmake can find it, you
+may need to add their installation paths to the `CMAKE_PREFIX_PATH` before running the steps
+above. For example:
+
+```bash
+$ export CMAKE_PREFIX_PATH=$CMAKE_PREFIX_PATH:/path/to/netcdf:/path/to/netcdf-fortran
+```
+
+### Building on a Mac
+
+By default, gcc points to the clang compiler on Mac.  To use the GNU compiler on Mac, depending
+on how the GNU compiler was installed, you may need to specify the C compiler name as gcc-$version.
+For example:
+
+```bash
+$ export CC=gcc-10
 ```
 
 ## Testing the kernel
 
-To run the test suite (from the build directory):
+First, set the OpenMP variables, including number of threads you want to use for the tests. For example:
 
-`ctest`
+```bash
+$ export OMP_PLACES=cores
+$ export OMP_PROC_BIND=close
+$ export OMP_NUM_THREADS=4
+```
+
+Then, to run the test suite (from the build directory):
+
+```bash
+$ ctest
+```
 
 To run a specific test (for example):
 
-`ctest -R regression_0.0.1`
+```bash
+$ ctest -R regression_12x24
+```
 
 To run a specific test with full output to get more information about a failure (for example):
 
-`ctest -VV -R regression_0.0.1`
+```bash
+$ ctest -VV -R regression_12x24
+```
 
 ## Build and test script
 
-For convenience, a build script is provided that builds the code and runs the test suite:
+For convenience, a build script is provided that builds the code and runs the test suite.
 
-**(NOTE: This script may need to be modified, depending on how modules are set up on your system)**
+**(NOTE: This script is written for machines that use modules and it may need to be modified,
+depending on how modules are set up on your system)**
 
-`./build.sh <gcc | intel> <debug | release>`
+```bash
+$ ./build.sh <gcc | intel> <debug | release>
+```
 
 ## Installation and running
 
 To (optionally) install the built executable into exe/
 
-`make install`
+```bash
+$ make install
+```
 
 To run the installed executable (for example):
 
-`exe/yppm ../test/test_input/yppm_0.0.1.nl`
+```bash
+$ export OMP_PLACES=cores
```

Comment:
This looks like c_sw, but yppm is single threaded.

Reply:
Same as before.  Remove the first two lines and keep `OMP_NUM_THREADS=1`